contracts/rollup/BatchManager.sol

```diff
@@ -127,14 +121,17 @@ contract BatchManager is Parameters {
                 block.number + paramBlocksToFinalise
             )
         });
-        batches.push(newBatch);
-        emit NewBatch(msg.sender, batches.length - 1, batchType);
+        emit NewBatch(msg.sender, nextBatchID, batchType);
+        nextBatchID++;
```

Comment:
A downside is we have to carefully track `nextBatchID`

Reply:
Do you want to create some internal methods like `add` that would take care of this automatically?

Reply:
We have 2 places that use nextBatchID++, should be manageable without the internal method for now.